edb/edgeql-rust/src/normalize.rs

```diff
@@ -7,11 +7,11 @@ use bigdecimal::BigDecimal;
 use crate::tokenizer::{CowToken, decode_string};
 
 
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
 pub enum Value {
     Str(String),
     Int(i64),
-    Float(f64),
+    Float(u64), // ... we store the raw bits here so that this can be Eq
```

Comment:
Let's make it `[u8; 8]` if we want to store bits. But I'm not sure if this is optimal.